src/snowflake/cli/_plugins/streamlit/streamlit_entity.py

```diff
@@ -1,12 +1,97 @@
+import functools
+import shutil
+
+from snowflake.cli._plugins.connection.util import make_snowsight_url
 from snowflake.cli._plugins.streamlit.streamlit_entity_model import (
     StreamlitEntityModel,
 )
-from snowflake.cli.api.entities.common import EntityBase
+from snowflake.cli._plugins.workspace.context import ActionContext
+from snowflake.cli.api.entities.common import EntityBase, get_sql_executor
+from snowflake.connector.cursor import SnowflakeCursor
 
 
 class StreamlitEntity(EntityBase[StreamlitEntityModel]):
```

Comment:
I added a temporary mock implementation of `StreamlitEntity` for composability with Native Apps in #1856.
The fake logic will be replaced by your PR, but please take a look at the proposed `ApplicationPackageChildInterface` contract.

Reply:
Renamed some methods to allign it with your interface.
Bundle and deploy actions are left intact, as they will have to be heavily remodeled, after introducing project-wide bundle map and other changes to output files management